Fog.js

```diff
@@ -465,53 +465,74 @@ function check_token_visibility(){
 		let auraSelectorId = $(".token[data-id='" + id + "']").attr("data-id").replaceAll("/", "");
 		let auraSelector = ".aura-element[id='aura_" + auraSelectorId + "']";
-		let selector = "div.token[data-id='" + id + "']";
+		let tokenSelector = "div.token[data-id='" + id + "']";
 
 
 		let playerTokenId = $(`.token[data-id*='${window.PLAYER_ID}']`).attr("data-id");
 		let playerTokenHasVision = (playerTokenId == undefined) ? true : window.TOKEN_OBJECTS[playerTokenId].options.auraislight;
 
 		//Combining some and filter cut down about 140ms for average sized picture
-		let someFilter = function(color, index) {return (index) % 4 == 0 && color == 255};
-		
-		if (!window.TOKEN_OBJECTS[id].options.revealInFog && (pixeldata[3] == 255 || (!pixeldata2.some(someFilter) && playerTokenHasVision && (window.CURRENT_SCENE_DATA.darkness_filter > 0 || window.walls.length>4)) || (playerTokenHasVision && window.CURRENT_SCENE_DATA.darkness_filter > 0  && (!is_token_under_light_aura(id) && pixeldata[2] == 0 && window.CURRENT_SCENE_DATA.darkness_filter > 0)))) {
-			$(selector + "," + auraSelector).hide();
+		let someFilter = function(ctx) {
+			let pixelData = ctx2.getImageData(parseInt(window.TOKEN_OBJECTS[id].options.left.replace('px', ''))/ window.CURRENT_SCENE_DATA.scale_factor, parseInt(window.TOKEN_OBJECTS[id].options.top.replace('px', ''))/ window.CURRENT_SCENE_DATA.scale_factor, window.TOKEN_OBJECTS[id].sizeWidth()/ window.CURRENT_SCENE_DATA.scale_factor, window.TOKEN_OBJECTS[id].sizeHeight()/ window.CURRENT_SCENE_DATA.scale_factor).data;
+			//return (index) % 4 == 0 && color == 255
+			for (let i = 0; i < pixelData.length; i+=4) {
+				if (pixelData[i] === 255) {
+					return true;
+				}
+			}
+
+			return false;
+		};
+
+		if (!window.TOKEN_OBJECTS[id].options.revealInFog 
+			&& (pixeldata[3] == 255
+				|| playerTokenHasVision 
+				&& ((window.CURRENT_SCENE_DATA.darkness_filter > 0  && !is_token_under_light_aura(id) && pixeldata[2] == 0 && window.CURRENT_SCENE_DATA.darkness_filter > 0))
+					|| ((window.CURRENT_SCENE_DATA.darkness_filter > 0 || window.walls.length>4) && !someFilter(ctx2)))) {
+			$(tokenSelector + "," + auraSelector).hide();
 		}
-		else if (!window.TOKEN_OBJECTS[id].options.hidden) {
-			let selectors = []
-			$(selector).css({'opacity': 1, 'display': 'block'});
+		else if (!window.TOKEN_OBJECTS[id].options.hidden ) {
+			$(tokenSelector).css({'opacity': 1, 'display': 'block'});
 			if(!window.TOKEN_OBJECTS[id].options.hideaura && id != playerTokenId)
 				$(auraSelector).show();
 			//console.log('SHOW '+id);
 		}
-	}
+	}, 700)));}

```

Comment:
Why the 700ms timeout for this or timeout at all really? Is this required for promises? I don't really know anything about them.

Reply:
Ah maybe this is how you're splitting them up?

Reply:
The setTimeout will just prevent them running for longer than 700 ms. If they do, it will kill it. But that isn't needed really with the awaits because control will return back to the user. Gonna remove